amy/autoemails/actions.py

```diff
@@ -1235,3 +1244,105 @@ def get_merge_data(self) -> Optional[Dict]:
     def reply_to(self) -> Optional[Tuple[str]]:
         """Overwrite in order to set own reply-to from descending Action."""
         return (settings.ADMIN_NOTIFICATION_CRITERIA_DEFAULT,)
+
+
+class ProfileUpdateReminderAction(BaseAction):
+    """
+    Action for asking users to login and update their profile information.
+    This email should be sent on the anniversary of the date the user's
+    profile was created.
+
+    This action is added during the RQ scheduler startup.
+    See amy.autoemails.management.commands.repeated_jobs
+    """
+
+    launch_at = timedelta(hours=1)
+
+    def recipients(self) -> Optional[Tuple[str]]:
+        """Assuming self.context is ready, overwrite email's recipients
+        with selected ones."""
+        try:
+            person_email = self.context_objects["person_email"]
+            return (person_email,)
+        except (KeyError, AttributeError):
+            return None
+
+    def all_recipients(self) -> str:
+        """If available, return string of all recipients."""
+        recipients = self.recipients()
+        if recipients is not None:
+            return ", ".join(recipients)
+        return ""
+
+    def reply_to(self) -> Optional[Tuple[str]]:
+        """Overwrite in order to set own reply-to from descending Action."""
+        return (settings.ADMIN_NOTIFICATION_CRITERIA_DEFAULT,)
+
+
+class BaseRepeatedAction:
+    """
+    Base class for repeated jobs in the Redis.
+    """
+
+    trigger: Trigger
+    launch_at: Optional[timedelta] = None
```

Comment:
I guess this is not useful for repeated actions, is it? Can we remove it? Would this break a lot of stuff?

Reply:
Removed